File: driver-scala/src/integration/scala/org/mongodb/scala/syncadapter/SyncListDatabasesIterable.scala
package org.mongodb.scala.syncadapter

import com.mongodb.client.ListDatabasesIterable
import org.bson.conversions.Bson
import org.mongodb.scala.ListDatabasesObservable
import org.mongodb.scala.bson.BsonValue

import java.util.concurrent.TimeUnit

case class SyncListDatabasesIterable[T](wrapped: ListDatabasesObservable[T])
    extends SyncMongoIterable[T]
    with ListDatabasesIterable[T] {
  override def maxTime(maxTime: Long, timeUnit: TimeUnit): ListDatabasesIterable[T] = {
    wrapped.maxTime(maxTime, timeUnit)
    this
  }

  override def batchSize(batchSize: Int): ListDatabasesIterable[T] = {
    wrapped.batchSize(batchSize)
    this
  }

  override def filter(filter: Bson): ListDatabasesIterable[T] = {
    wrapped.filter(filter)
    this
  }

  override def nameOnly(nameOnly: java.lang.Boolean): ListDatabasesIterable[T] = {
    wrapped.nameOnly(nameOnly)
    this
  }

  override def authorizedDatabasesOnly(authorizedDatabasesOnly: java.lang.Boolean): ListDatabasesIterable[T] = {
    wrapped.authorizedDatabasesOnly(authorizedDatabasesOnly)
    this
  }

  override def comment(comment: String): ListDatabasesIterable[T] = {
    wrapped.comment(comment)
    this
  }

  override def comment(comment: BsonValue): ListDatabasesIterable[T] = {
    wrapped.comment(comment)
    this
  }
}

File: driver-kotlin-sync/src/integration/kotlin/com/mongodb/kotlin/client/syncadapter/SyncListDatabasesIterable.kt
/*
 * Copyright 2008-present MongoDB, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.mongodb.kotlin.client.syncadapter

import com.mongodb.client.ListDatabasesIterable as JListDatabasesIterable
import com.mongodb.kotlin.client.ListDatabasesIterable
import java.util.concurrent.TimeUnit
import org.bson.BsonValue
import org.bson.conversions.Bson

data class SyncListDatabasesIterable<T>(val wrapped: ListDatabasesIterable<T>) :
    JListDatabasesIterable<T>, SyncMongoIterable<T>(wrapped) {

    override fun batchSize(batchSize: Int): SyncListDatabasesIterable<T> = apply { wrapped.batchSize(batchSize) }

    override fun maxTime(maxTime: Long, timeUnit: TimeUnit): SyncListDatabasesIterable<T> = apply {
        wrapped.maxTime(maxTime, timeUnit)
    }

    override fun filter(filter: Bson?): SyncListDatabasesIterable<T> = apply { wrapped.filter(filter) }

    override fun nameOnly(nameOnly: Boolean?): SyncListDatabasesIterable<T> = apply { wrapped.nameOnly(nameOnly) }

    override fun authorizedDatabasesOnly(authorizedDatabasesOnly: Boolean?): SyncListDatabasesIterable<T> = apply {
        wrapped.authorizedDatabasesOnly(authorizedDatabasesOnly)
    }

    override fun comment(comment: String?): SyncListDatabasesIterable<T> = apply { wrapped.comment(comment) }

    override fun comment(comment: BsonValue?): SyncListDatabasesIterable<T> = apply { wrapped.comment(comment) }
}
